pkg/core/consensus/user/sortition.go

```diff
@@ -126,7 +126,7 @@ func (p Provisioners) CreateVotingCommittee(round uint64, step uint8, size int)
 func (p Provisioners) extractCommitteeMember(score uint64) []byte {
 	for i := 0; ; i++ {
 		// make sure we wrap around the provisioners array
-		if i >= len(p.Members) {
+		if i >= len(p.Set) {
```

Comment:
Usually it's better to have this bound check in `MemberAt` returning error to avoid any panics on another use of the method.

Reply:
That's a good point. Will add it

Reply:
I too find more idiomatic if `MemberAt` actually returns an error when the client is asking for a member at an index inconsistent with the set size. It is up to the client to handle such error, rather than `MemberAt` to silently return the first member, which is an arbitrary result to the client's query

Reply:
I still think that the `extractCommitteeMember` should panic in case the handling fails (since it means that there are no members in the Committee, which should never happen)